---
title: "Final_Project"
author: "Arad Peron & Omer Arbel"
date: "2024-05-18"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
#Arad:
setwd('~/Library/Mobile Documents/com~apple~CloudDocs/Technion/SemesterC/Bioinfo/Final Project/')

#Omer:
#setwd('~/Desktop/Technion/semesterC/Bioinformatics/')
```

```{r}
library(statgenGWAS)

library(tidyverse)
library(qqman)
```
Preparing data:
```{r}
GWASResults <- read_tsv("GannaA_prePMID_MTAG_nonhetero_cc_MF.tsv")
GWASResults_reduce <- GWASResults %>%
  dplyr::select(SNP, CHR, BP, PVAL) %>%
  dplyr::rename(P = PVAL) %>%
  dplyr::filter(!is.na(P))

```


QQ-plot:
```{r}
qq(GWASResults_reduce$P, main = "Q-Q plot of GWAS p-values", col = "blue4")
```

As in the plot above, deviations from this line should only occur on the right side of the plot, for a small number of SNPs with an effect on the phenotype (and possibly SNPs in linkage disequilibrium (LD)).


Manhattan plot:
```{r}
show(GWASResults_reduce)
manhattan(GWASResults_reduce, main = "Manhattan Plot", ylim = c(0, 10), 
          cex = 0.6, cex.axis = 0.9, col = c("blue4", "orange3"), 
          suggestiveline = FALSE, genomewideline = FALSE, 
          chrlabs = c(1:20, "P", "Q"))
```


- Highly your favorit SNPs
```{r}
snpsOfInterest <- c("rs9442400")
```

```{r}
manhattan(GWASResults_reduce, highlight = snpsOfInterest)
```

```{r}
manhattan(subset(gwasResults_reduce, CHR == 7), highlight = snpsOfInterest, xlim = c(44800000, 45000000))
```